sys/freebsd/sys.txt

```diff
@@ -175,7 +184,49 @@ pollfd {
 }
 
 sigset {
-	mask	int64
+	mask	array[int32, _SIG_WORDS]
+}
+
+siginfo {
+	si_signo	int32
+	si_errno	int32
+	si_code		int32
+	si_pid		pid
+	si_uid		uid
+	si_status	int32
+	si_addr		ptr[in, array[int8]]
```

Comment:
I think `intptr` makes more sense for `si_addr`. siginfo is always exported by the kernel, and `si_addr` just points to the text address of the interrupted thread.